File: 1666-make-the-string-great/make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char> st;
        for(char ch:s){
            if(!st.empty() && isBad(st.top(),ch)){
                st.pop();
            }
            else{
                st.push(ch);
            }
        }
        string sb;
        while(!st.empty()){
            sb+=st.top();
            st.pop();
        }
        reverse(sb.begin(),sb.end());
        return sb;
    }
    bool isBad(char a,char b){
        return tolower(a)==tolower(b) && a!=b;
    }
};

File: 1666-make-the-string-great/make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        Stack<Character> st=new Stack<>();
        for(char ch:s.toCharArray()){
            if(!st.isEmpty() && isBad(st.peek(),ch)){
                st.pop();
            }
            else{
                st.push(ch);
            }
        }
        StringBuilder sb=new StringBuilder();
        while(!st.isEmpty()){
            sb.append(st.pop());
        }
        return sb.reverse().toString();
    }
    public boolean isBad(char a,char b){
        return a!=b && Character.toLowerCase(a)==Character.toLowerCase(b);
    }
}